src/Lean/Elab/Tactic/Simp.lean

```diff
@@ -263,6 +270,14 @@ def traceSimpCall (stx : Syntax) (usedSimps : UsedSimps) : MetaM Unit := do
       if env.contains declName && !simpOnlyBuiltins.contains declName then
         args := args.push (← `(Parser.Tactic.simpLemma| $(mkIdent (← unresolveNameGlobal declName)):ident))
     | .fvar fvarId => -- local hypotheses in the context
+      if isSimpAll then
+        continue
```

Comment:
`simp_all` only uses nondependent hypotheses.

Reply:
If I read `initEntries` right, it uses all hypotheses but only adds nondependent ones as entries to simplify. However, trying to test this, I'm getting slightly mysterious (to me) behaviour:

```lean
example {P : Nat → Type} {Q : (n m : Nat) → n = m → Type} (h₁ : n = m) (h₂ : P n) : Q n m h₁ := by
  simp_all
  /-
  ...
  h₁ : n = m
  h₂ : P n
  ⊢ Q m m (_ : m = m)
  -/
```

So it used the dependent `h₁` to rewrite in the target, but not in the nondependent `h₂`. Is this intended? Maybe it's just getting too late for me...